rust/main/hyperlane-base/src/server/base_server.rs

```diff
@@ -41,13 +41,16 @@ impl Server {
             app = app.nest(route, router);
         }
 
-        tokio::spawn(async move {
-            let addr = SocketAddr::from(([0, 0, 0, 0], port));
-            axum::Server::bind(&addr)
-                .serve(app.into_make_service())
-                .await
-                .expect("Failed to start server");
-        })
+        tokio::task::Builder::new()
+            .name("agent::server")
+            .spawn(async move {
+                let addr = SocketAddr::from(([0, 0, 0, 0], port));
+                axum::Server::bind(&addr)
+                    .serve(app.into_make_service())
+                    .await
+                    .expect("Failed to start server");
+            })
+            .expect("spawning tokio task from Builder is infalliable")
```

Comment:
nit, also in a couple other places
```suggestion
            .expect("spawning tokio task from Builder is infallible")
```

Reply:
Fixed